flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/dml/RichSqlInsert.java

```diff
@@ -93,6 +97,41 @@ public SqlNodeList getStaticPartitions() {
 		getSource().unparse(writer, 0, 0);
 	}
 
+	//~ Tools ------------------------------------------------------------------
+
+	public static boolean isUpsert2(List<SqlLiteral> keywords) {
+		for (SqlNode keyword : keywords) {
+			RichSqlInsertKeyword keyword2 =
+				((SqlLiteral) keyword).symbolValue(RichSqlInsertKeyword.class);
+			if (keyword2 == RichSqlInsertKeyword.UPSERT) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/** Returns whether this is an UPSERT. */
+	public boolean isUpsert2() {
+		return getModifierNode2(RichSqlInsertKeyword.UPSERT) != null;
+	}
+
+	/** Returns whether the insert mode is overwrite
+	 * (for whole table or for specific partitions). */
+	public boolean isOverwrite() {
+		return getModifierNode2(RichSqlInsertKeyword.OVERWRITE) != null;
+	}
+
+	private SqlNode getModifierNode2(RichSqlInsertKeyword modifier) {
```

Comment:
just name it with `getModifierNode`?

Reply:
There is already a method named `getModifierNode` in super class